### ISSUE TITLE
Incorporate NodePort into minishift openshift service

### ISSUE THREAD

**Issue by praveenkumar** [minishift/minishift#1106]
This is a follow up issue for #876.

The current behaviour is to open the route of a service in the default browser, provided one is defined. Is this what we want? 

Should we change this  so that when a user asks for a specific service then we display all associated routes and nodeports. To open in browser  `--inbrowser` needs to be explicitly specified.

Something like

```
$ minishift openshift service guestbook-v2 -n myproject
|-----------------|----------------------|--------------------------------------------------------|--------|
|      NAME       |       NODEPORT       |                       ROUTE-URL                        | WEIGHT |
|-----------------|----------------------|--------------------------------------------------------|--------|
| guestbook-v2    |                      | http://guestbook-v2-3002-myproject.192.168.64.2.nip.io |  50%   |
|                 |                      | http://guestbook-myproject.192.168.64.2.nip.io         |        |
|                 |                      | http://guestbook-v2-myproject.192.168.64.2.nip.io      |        |
|-----------------|----------------------|--------------------------------------------------------|--------|

$ minishift openshift service guestbook-v2 -n myproject --inbrowser
< Open a routes in browser>
```

Also in the light of that a service can have multiple routes, how do we deal with that?

@jorgemoralespou  asked openshift folks about how to display the route and they provided below explanation but I think for our minishift purpose we don't have to put all in to account and just open first from index, let me know if that is wrong.

*In the console we use a scoring algorithm that takes several factors into account https://github.com/openshift/origin-web-console/blob/master/app/scripts/services/routes.js#L106*

**Comment by hferentschik** [comment]
@minishift/minishift-dev, @jorgemoralespou, @jstrachan - thoughts on how `minishift openshift service [name]` should behave.

**Comment by jorgemoralespou** [comment]
@hferentschik I find it ok, although I might think that not too much people will use it. I would probably use this instead:

````
oc get route/guestbok-2 -o jsonpath='{.spec.host}'
````

If I want to script the value, or else:

````
oc get routes
````

And as item2 gives me click on a url and open in a browser, I would just click on the URL. I would not need the --inbrowser --index 2, which is too much to learn IMHO in case I just want to open a route.

Long answer short. Too many options to get to a route ;-)

**Comment by hferentschik** [comment]
I tend to agree, for the single route case your oc command is probably the faster/easier approach. However you are also a power user and other less advanced users might not be so experienced. For example the use of jsonpath is not so trivial. It took me a while to understand it.

Also it is about providing a consistent user experience. Not all commands need imo to be better/faster than the alternative approach via oc or even the console. Whether people will use it is another question. The `service list` command has for sure more value. You would have to run a whole bunch of _oc_ commands to get this information. This task is mainly about making things coherent.

**Comment by jorgemoralespou** [comment]
That's mostly why I said I find it ok. Not sure how often people will use it until we see people using it, so I'm ok.

**Comment by hferentschik** [comment]
BTW, using an index requires that we can guarantee an order in which we display routes. Not sure whether this is required or whether we can start with just opening the first one. We can then wether the need evert arises for something else.

**Comment by LalatenduMohanty** [comment]
> thoughts on how minishift openshift service [name] should behave.

I agree with what is there in the original description except I do not think we should put effort to make --inbrowser work. I think I have told it before  that I do not see much use of it. So why put  lot efforts now. Rather we can have a followup issue and if we some interest we can do it. It is also not critical from my point of view.

**Comment by hferentschik** [comment]
Merged via pull request #1114
